### ISSUE TITLE
Nested double quotes inside date format not caught as invalid

### ISSUE THREAD

**Issue by meramsey** [koalaman/shellcheck#2553]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
/bin/bash /home/ubuntu/gitlab_backup.sh 2>&1 | tee -a "/home/ubuntu/backup_logs/$(date +"%Y-%m-%d")_backup_cron.log"; 

```

#### Here's what shellcheck currently says:
```

```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

Only found this out after the cron entry failed and I noticed the split.
```
root@coolserver:~# sudo grep -is root /var/log/{cron*,syslog*}| grep CRON | grep 'gitlab_backup.sh'
/var/log/syslog.1:Aug 11 01:00:01 gitlab1 CRON[4372]: (root) CMD (/bin/bash /home/ubuntu/gitlab_backup.sh 2>&1 | tee -a "/home/ubuntu/backup_logs/$(date +")
root@coolserver:~# 
```

**Comment by koalaman** [comment]
The command is fine as written, since `$(..)` starts a new quoting context. However, if you were to copy-paste it from the script into your crontab, cron will terminate the command at the first unescaped `%` and treat the rest as stdin. (For example, `0 0 * * * nl > file %foo%bar%baz` will run `nl > file` and feed the lines `foo`, `bar`, and `baz` on stdin).

It's certainly a problem, but specific to `cron`'s processing of the command before giving it to the shell. If you put the line in a script and invoke the script via crontab, it will work as expected.
